index.js

```diff
@@ -1051,40 +1061,46 @@
       };
     };
 
-    //  _determineActualTypes :: Boolean -> [Any] -> [Type]
-    var _determineActualTypes = function(loose) {
-      return function recur(values) {
-        if (isEmpty(values)) return [Unknown];
-        //  consistentTypes :: [Type]
-        var consistentTypes = chain(env, rejectAny);
-        //  typeses :: [[Type]]
-        var typeses = map(values, function(value) {
-          return chain(consistentTypes, function(t) {
-            return (
-              t.name === 'sanctuary-def/Nullable' || !test(t, value).valid ?
-                [] :
-              t.type === 'UNARY' ?
-                map(recur(t._1(value)), UnaryType.from(t)) :
-              t.type === 'BINARY' ?
-                BinaryType.xprod(t, recur(t._1(value)), recur(t._2(value))) :
-              // else
-                [t]
-            );
-          });
-        });
-        //  common :: [Type]
-        var common = commonTypes(typeses, loose);
-        if (!isEmpty(common)) return common;
-        //  If none of the values is a member of a type in the environment,
-        //  and all the values have the same type identifier, the values are
-        //  members of a "foreign" type.
-        if (isEmpty(filterTypesByValues(consistentTypes, values)) &&
-            all(values.slice(1), $$typeEq($$type(values[0])))) {
-          //  Create a nullary type for the foreign type.
-          return [type0($$type(values[0]))];
+    //  _determineActualTypes :: (Boolean, [Object], [Any]) -> [Type]
+    var _determineActualTypes = function recur(loose, $seen, values) {
```

Comment:
This is a [named function expression][1].

> If you want to refer to the current function inside the function body, you need to create a named function expression. This name is then local only to the function body (scope). This also avoids using the non-standard [`arguments.callee`][2] property.


[1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/function#Named_function_expression
[2]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments/callee